resolve.go

```diff
@@ -45,29 +45,13 @@ func Resolve(err error) error {
 func firstError(err error) error {
 	for {
 		switch err {
-		case ErrUnknown,
-			ErrInvalidArgument,
-			ErrNotFound,
-			ErrAlreadyExists,
-			ErrPermissionDenied,
-			ErrResourceExhausted,
-			ErrFailedPrecondition,
-			ErrConflict,
-			ErrNotModified,
-			ErrAborted,
-			ErrOutOfRange,
-			ErrNotImplemented,
-			ErrInternal,
-			ErrUnavailable,
-			ErrDataLoss,
-			ErrUnauthenticated,
-			context.DeadlineExceeded,
-			context.Canceled:
+		case context.Canceled, context.DeadlineExceeded:
 			return err
 		}
+
 		switch e := err.(type) {
-		case customMessage:
-			err = e.err
+		case Error:
```

Comment:
Should keep the custom error case rather than rely on Unwrap

Reply:
I suspect that keeping `customMessage` here when the more traditional `Unwrap()` works might be an anti-pattern and creates the possibility of more dead code.

Is there a reason why it would be better to keep `customMessage` here rather than rely on `Unwrap()`?